gitpod-io/gitpod
# [agent-smith] Use proc scanning

## Description
This PR changes agent-smith from BPF over to proc scanning. The latter is easier to understand and operate, while still providing a useful response time.

Inspired by and replacement for #6018

## Related Issue(s)

Fixes #6138, #6136

## How to test

In one workspace build the test target
```
leeway build --serve=:8080 components/ee/agent-smith/cmd/testtarget:app
```

In a workspace started on this branch, download the test target and execute it
```bash
curl -OL <wsurl>/testtarget
chmod +x testtarget
./testtarget
```

Your workspace should have been stopped.

## Release Notes

```release-note
NONE
```

## Documentation
